pkg/schemaregistry/service.go

```diff
@@ -14,17 +14,61 @@
 
 package schemaregistry
 
-//go:generate mockgen -destination=mock/schema_service.go -package=mock -mock_names=Service=Service . Service
-
 import (
 	"context"
 
-	commschema "github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit-connector-protocol/pconduit"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/schemaregistry/fromschema"
+	"github.com/conduitio/conduit/pkg/schemaregistry/toschema"
+	"github.com/twmb/franz-go/pkg/sr"
 )
 
-type Service interface {
-	Create(ctx context.Context, name string, bytes []byte) (commschema.Instance, error)
-	Get(ctx context.Context, id string) (commschema.Instance, error)
+type Service struct {
+	registry Registry
+	logger   log.CtxLogger
+}
+
+var _ pconduit.SchemaService = (*Service)(nil)
+
+func NewService(logger log.CtxLogger, registry Registry) *Service {
+	return &Service{
+		registry: registry,
+		logger:   logger,
+	}
+}
+
+func (s *Service) Check(ctx context.Context) error {
+	r, ok := s.registry.(RegistryWithCheck)
+	if !ok {
+		return nil
+	}
+	return r.Check(ctx)
+}
+
+func (s *Service) CreateSchema(ctx context.Context, req pconduit.CreateSchemaRequest) (pconduit.CreateSchemaResponse, error) {
+	ss, err := s.registry.CreateSchema(ctx, req.Subject, sr.Schema{
+		Schema: string(req.Bytes),
+		Type:   fromschema.SrSchemaType(req.Type),
+	})
+	if err != nil {
+		// TODO: convert error
+		return pconduit.CreateSchemaResponse{}, cerrors.Errorf("failed to create schema: %w", err)
+	}
+	return pconduit.CreateSchemaResponse{
+		Schema: toschema.SrSubjectSchema(ss),
+	}, nil
+}
+
+func (s *Service) GetSchema(ctx context.Context, req pconduit.GetSchemaRequest) (pconduit.GetSchemaResponse, error) {
+	ss, err := s.registry.SchemaBySubjectVersion(ctx, req.Subject, req.Version)
+	if err != nil {
+		// TODO: convert error
```

Comment:
This PR or another one?

Reply:
This one for sure 😬

Reply:
I changed the conduit schema registry to [return `sr.ResponseError`](https://github.com/ConduitIO/conduit-schema-registry/pull/2) same as the client, then convert the errors accordingly. We need to test it end-to-end to know if it works as expected. I'll add unit tests separately.